recipes/libheif/all/patches/0001-cmake.patch

```diff
@@ -0,0 +1,47 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 5076fec..b75be8e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -41,8 +41,6 @@ set(CMAKE_CXX_EXTENSIONS OFF)
+ set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+ set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
+ 
+-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+-
+ # Create the compile command database for clang by default
+ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+ 
+@@ -55,8 +53,8 @@ if (has_potentially_evaluated_expression)
+ endif()
+ 
+ LIST (APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules")
+-find_package(Libde265 REQUIRED)
+-find_package(X265)
++find_package(libde265 REQUIRED)
++find_package(libx265)
```

Comment:
Does upstream not follow the official targets or are they incorrect from C3i?

Reply:
1) Libde265 is incorrect because upper-case `LIBDE265_FOUND` is used afterwards. It should be rechecked and sent upstream as a patch (just to lowercase everything).

2) x265 Conan recipe does not produce `cmake_find_package` names so it is `libx265` by default. Upstream is not specific about cmake package name but I think it should be `x265` without `lib` prefix as in pkg-config.

Reply:
2a. Filled a PR with changing cmake name for libx265 package (#3433) - @madebr , please take a look.
Anyway, patches are needed until upstream would change dep names to lowercase.

Reply:
You have 2 solutions:
- patch libheif CMakeLists to consume conan targets
- use pkg_config generator (+ add pkgconf in build_requirements()) because the custom X265 module file in libheif rely on official X265 pkg-config files: https://github.com/strukturag/libheif/blob/master/cmake/modules/FindX265.cmake

Reply:
About (2a) - decided in 3433 that `libx265` cmake name should be used so patching of upstream CMakeLists is required.

So solution 1 is preferable. Upstream finders are dropped and only conan-generated finders are used.

Reply:
As I said, you have a third solution: pkg-config generator (advantage: no patch, drawback: it adds a `pkgconf` in `build_requirements()`)